src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -894,7 +894,8 @@ def add_snowpark_package_to_sproc_packages(
         elif package_name not in session._packages:
             packages = list(session._packages.values()) + [this_package]
     else:
-        if not any(package_name in p for p in packages):
+        package_names = [p if isinstance(p, str) else p.__name__ for p in packages]
+        if not any(p.startswith(package_name) for p in package_names):
```

Comment:
Will be removed from this PR once https://github.com/snowflakedb/snowpark-python/pull/1401 is merged